### PR TITLE
Set DNSPolicy=ClusterFirstWithHostNet when hostnetwork set

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
https://github.com/volcano-sh/volcano/issues/778

### CODE REVIEW

File: pkg/controllers/job/job_controller_util.go [hunk 1]
```diff
@@ -57,6 +57,10 @@ func createJobPod(job *batch.Job, template *v1.PodTemplateSpec, ix int) *v1.Pod
 	if len(pod.Spec.SchedulerName) == 0 {
 		pod.Spec.SchedulerName = job.Spec.SchedulerName
 	}
+	// If dns policy not set when hostNetwork=true, set it to `ClusterFirstWithHostNet`
+	if pod.Spec.HostNetwork == true && pod.Spec.DNSPolicy == "" {
```
Comment (k82cn) [inline pkg/controllers/job/job_controller_util.go:61]: can we add this into admission controller?

File: pkg/controllers/job/job_controller_util.go [hunk 2]
```diff
@@ -57,6 +57,10 @@ func createJobPod(job *batch.Job, template *v1.PodTemplateSpec, ix int) *v1.Pod
 	if len(pod.Spec.SchedulerName) == 0 {
 		pod.Spec.SchedulerName = job.Spec.SchedulerName
```
Comment (k82cn) [inline pkg/controllers/job/job_controller_util.go:58]: ditto, admission controller.

File: pkg/controllers/job/job_controller_util.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_util.go:61]: Yes, i must be drunk

Comment (hzxuzhonghu) [reply pkg/controllers/job/job_controller_util.go:61]: yes, anyway makes sense to me

File: pkg/webhooks/admission/jobs/mutate/mutate_job.go [hunk 3]
```diff
@@ -33,8 +33,10 @@ import (
 )
 
 const (
-	//DefaultQueue constant stores the name of the queue as "default"
+	// DefaultQueue constant stores the name of the queue as "default"
 	DefaultQueue = "default"
+
+	defaultSchedulerName = "volcano"
```
Comment (k82cn) [inline pkg/webhooks/admission/jobs/mutate/mutate_job.go:39]: Let's move this as an argument/parameter, similar to `DefaultQueue`.

**Pull request merged**